Integrate weight field of ProductVariantVersion

lesser1 opened the pull request:
- Add weight information to ProductView
    - Add Unit enum for units like Kilogram
    - Store unit used for weight information in app store,
      have Kilogram as default value
    - Add textfield to AddProductDialog that for specifying
      weight of product variant
    - Add rules.ts file with common functions that can be used
      for the rules props of vue SFCs like v-textfield
    - Add weight field to product queries
    - Reorder queried fields of ProductVariantVersion

[The Gropius Issue](https://frontend.gropius.duckdns.org/components/9d12d6ac-e85a-407c-bdff-a8321fb75d3a/issues/8ab30955-b180-4bcc-8b75-dde011b94a19)

### Definition of Done
- [x] Requirements of the issue are met
- [x] Code has been reviewed
- [x] Code is documented
- [x] GraphQL related artefacts are documented

lesser1 committed:
Integrate weight field of ProductVariantVersion:

    - Add weight information to ProductView
    - Add Unit enum for units like Kilogram
    - Store unit used for weight information in app store,
      have Kilogram as default value
    - Add textfield to AddProductDialog that for specifying
      weight of product variant
    - Add rules.ts file with common functions that can be used
      for the rules props of vue SFCs like v-textfield
    - Add weight field to product queries
    - Reorder queried fields of ProductVariantVersion

nk-coding commented on src/views/ProductView.vue:
```diff
@@ -275,7 +294,15 @@
+                                <v-col>{{
+                                    productVariant?.currentVersion.weight +
+                                    ' ' +
+                                    unitUsedForWeightInformation
```
imho this does not work
the backend requires this stuff to be saved in kg
if you want to support others, you need to do unit conversion yourself

lesser1 replied:
Oh, okay. Didn't know its kg in the backend. I will change it to kg only.

nk-coding commented on src/store/app.ts:
```diff
@@ -35,6 +37,7 @@ export const useAppStore = defineStore('app', {
         userRolesOfCurrentUser: initialUserRolesOfCurrentUser,
         activeUserRole: defaultUserRole,
         queuedNotifications: [] as Notification[],
+        unitUsedForWeightInformation: defaultUnitUsedForWeightInformation,
```
this might be a candidate to be saved in localstorage

delvh replied:
Additionally, I would prefer a shorter name like `weightUnit`

nk-coding commented on src/util/unit.ts:
```diff
@@ -0,0 +1,8 @@
+ * @enum
+ */
+export enum Unit {
+    Kilogram = 'kg',
```
if you want to support different units in the future, this would need to support unit conversion

lesser1 replied:
Since it's kg in the backend, I will keep it simple and only support kg.

nk-coding reviewed (commented):
Some minor things, not really a request changes, but also not really an approve.
Will wait for answers and probably approve then

delvh commented on src/util/rules.ts:
```diff
@@ -0,0 +1,34 @@
+ * @returns True if the weight is valid, otherwise a string describing the error.
+ */
+export function weightInputIsValid(input: string): boolean | string {
+    if (isNotNull(input) && isNumber(input) && parseFloat(input) > 0) {
```
```suggestion
    if (isNumber(input) && parseFloat(input) > 0) {
```

delvh reviewed (commented):
Same as @nk-coding

nk-coding commented:
What's the state on this?

lesser1 commented:
> What's the state on this?

I'm on it, changes are in progress.

lesser1 committed:
Update src/util/rules.ts

Simplify rule usage.

Co-authored-by: delvh <leon@kske.dev>

lesser1 committed:
Small changes:

    - Rename enum Unit to UnitOfMass.
    - Add state variable unitOfMass to local storage and rename it.
    - Ensure unitOfMass in AddProductDialog is always kg.

nk-coding commented on src/util/unitOfMass.ts:
```diff
@@ -0,0 +1,7 @@
+ * Represents units of mass, e.g. Kilogram, the base unit of mass.
+ * @enum
+ */
+export enum UnitOfMass {
```
if you only want to support kg, what's the point of this?

lesser1 replied:
Yeah, you're right.. I took care of it / removed it.

lesser1 committed:
Remove UnitOfMass and introduce global strings:

    - Add commonStrings.ts with frequentyl used strings.
    - Remove UnitOfMass enum and fix references to it.
